content/ch-gates/oracles.ipynb

```diff
@@ -0,0 +1,289 @@
+      "text/plain": [
+       "<Figure size 423.808x204.68 with 1 Axes>"
+      ]
+     },
+     "execution_count": 25,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "qc = Uf + Vf.inverse()\n",
+    "\n",
+    "qc.draw(output='mpl',justify='none')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This circuit begins with two identical `cx` gates, whose effects cancel each other out. All that remains is the final `cx` between the input and garbage registers. Mathematically, this means\n",
+    "\n",
+    "$$\n",
+    "V_f^\\dagger U_f \\left| x,f(x),0 \\right\\rangle = \\left| x , 0 ,g(x) \\right\\rangle.\n",
+    "$$\n",
+    "\n",
+    "This does not leave us with the state we require, but instead leaves the first qubit entangled with unwanted garbage. Any subsequent steps in an algorithm will therefore not run as expected, since the state is not the one that we need.\n",
+    "\n",
+    "For this reason we need a way of removing classical garbage from our quantum algorithms. This can be done by a method known as 'uncomputation. We simply need to take another blank variable and apply $V_f$\n",
+    "\n",
+    "\n",
+    "$$\n",
+    "\\left| x, 0, 0, 0 \\right\\rangle \\rightarrow \\left| x,f(x),g(x),0 \\right\\rangle.\n",
```

Comment:
Maybe, if there is a circuit for this uncomputation part as well, it would be helpful for reader to understand better...
Just personal opinion. :)